src/Rector/Class_/ProphecyPHPDocRector.php

```diff
@@ -160,14 +162,17 @@ private function findAssignToPropertyName(Class_ $class, string $propertyName):
 
     private function changePropertyDoc(Property $property, string $prophesizedClass): void
     {
-        $doc = new Doc(
-            \sprintf(
-                "/**\n     * @var %s<%s>\n     */",
-                '\Prophecy\Prophecy\ObjectProphecy',
-                '\\' . $prophesizedClass,
-            )
-        );
-
-        $property->setDocComment($doc);
+        $propertyPhpDocInfo = $this->phpDocInfoFactory->createFromNodeOrEmpty($property);
+
+        // replace old one
+        $propertyPhpDocInfo->removeByType(VarTagValueNode::class);
+
+        $propertyPhpDocInfo->addTagValueNode(new VarTagValueNode(
+            new GenericTypeNode(new FullyQualifiedIdentifierTypeNode('Prophecy\Prophecy\ObjectProphecy'), [
+                new FullyQualifiedIdentifierTypeNode($prophesizedClass),
+            ]),
+            '',
+            ''
+        ));
     }
 }
```

Comment:
@alexander-schranz Just a tip for work with docblocks :) 
It's better to use object API with PhpDocInfo rather than strings. Rector and PHPStan can manage it better and work with those as types :+1:

Reply:
Thx for the hint 👍